### PR TITLE
Add command aliases from Cargo to x.py commands

### LINKED ISSUE #71357: x.py feature parity with cargo (alias)
`cargo` currently supports `cargo t`, `cargo b` and a few others even though not so discoverable (https://github.com/rust-lang/cargo/issues/6104#issuecomment-589945732).

It would be good to let `x.py` supports the aliases supported by `cargo`, such as `./x.py t`, `./x.py b` and such.

This issue has been assigned to @mibac138 via [this comment](https://github.com/rust-lang/rust/issues/71357#issuecomment-623744646).

### PR DESCRIPTION
**Pull request by mibac138**
Fixes #71357

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
I'm not opposed to this though not really a fan myself, but it's a minimal patch set so @bors r+ rollup

File: src/bootstrap/flags.rs [hunk 1]
```diff
@@ -106,18 +106,18 @@ impl Flags {
 Usage: x.py <subcommand> [options] [<paths>...]
 
 Subcommands:
-    build       Compile either the compiler or libraries
-    check       Compile either the compiler or libraries, using cargo check
+    build, b    Compile either the compiler or libraries
```
Comment (pickfire) [inline src/bootstrap/flags.rs:109]: Doesn't `cargo` does not have the additional `, b`?

Comment (mibac138) [reply src/bootstrap/flags.rs:109]: No, it doesn't, but I think it should. I added it to promote discoverability.

Comment (pickfire) [reply src/bootstrap/flags.rs:109]: I think we can promote discoverability with `(b)uild` too.

Comment (pickfire) [reply src/bootstrap/flags.rs:109]: > No, it doesn't, but I think it should. I added it to promote discoverability.

Let me go add it.

**Pull request merged**